File: software/examples/javascript/ExampleSimple.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'XYZ'; // Change XYZ to the UID of your Solid State Relay Bricklet

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var ssr = new Tinkerforge.BrickletSolidStateRelay(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function (error) {
        console.log('Error: ' + error);
    }
); // Connect to brickd
// Don't use device before ipcon is connected

ipcon.on(Tinkerforge.IPConnection.CALLBACK_CONNECTED,
    function (connectReason) {
        // Turn relay on/off 10 times with 1 second delay
        for(var i = 0; i < 5; ++i) {
            setTimeout(function () {
                ssr.setState(true);
            }, 2000 * i + 1000);
            setTimeout(function () {
                ssr.setState(false);
            }, 2000 * i + 2000);
        }
    }
);

console.log('Press key to exit');
process.stdin.on('data',
    function (data) {
        ipcon.disconnect();
        process.exit(0);
    }
);

File: software/examples/go/example_simple.go
package main

import (
	"fmt"
	"github.com/Tinkerforge/go-api-bindings/ipconnection"
	"github.com/Tinkerforge/go-api-bindings/solid_state_relay_bricklet"
	"time"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your Solid State Relay Bricklet.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	ssr, _ := solid_state_relay_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	// Turn relay on/off 10 times with 1 second delay
	for i := 0; i < 5; i++ {
		time.Sleep(1000 * time.Millisecond)
		ssr.SetState(true)
		time.Sleep(1000 * time.Millisecond)
		ssr.SetState(false)
	}

	fmt.Print("Press enter to exit.")
	fmt.Scanln()

}
